lib/spring/client/binstub.rb

```diff
@@ -38,7 +38,7 @@ class Binstub < Command
   require 'bundler'
 
   if (match = Bundler.default_lockfile.read.match(/^GEM$.*?^    (?:  )*spring \((.*?)\)$.*?^$/m))
-    Gem.paths = { 'GEM_PATH' => [Bundler.bundle_path.to_s, *Gem.path].uniq }
+    Gem.paths = { 'GEM_PATH' => [Bundler.bundle_path.to_s, *Gem.path].uniq.join(Gem.path_separator) }
```

Comment:
Right, I was following from https://github.com/rubygems/rubygems/commit/27a436e91f1d83dc4a23168851d4d29a73fde752

but later it was indeed changed in https://github.com/rubygems/rubygems/commit/544d6fb3475b121a52eec79ab3df8877a2285387

Using `Gem.use_paths` seems like a best way, per @tenderlove comment in https://github.com/rubygems/rubygems/pull/1094#issuecomment-65871904

I'll create PR